spec/rest_in_peace/template_sanitizer_spec.rb

```diff
@@ -120,12 +120,20 @@
 
       context 'param which need to be encoded' do
         let(:params) { { id: 'this param need to be encoded' } }
-        let(:encoded_param) { 'this+param+need+to+be+encoded' }
+        let(:encoded_param) { 'this%20param%20need%20to%20be%20encoded' }
```

Comment:
Hello @cimnine 
First let me explain the problem I am facing.
I have a GET api endpoint where I want to pass value as part of the URL path like following:
`api/countries/CA/places/St Catharines`
RIP will escape the white space in the value with `+`, `St Catharines` => `St+Catherines`.
When the request reaches my application, the `+` sign will go unencoded as [rfc1738#section-2.2 defines](https://tools.ietf.org/html/rfc1738#section-2.2)
> only alphanumerics, the special characters "$-_.+!*'(),", and
   reserved characters used for their reserved purposes may be used
   unencoded within a URL.

On the other hand if I pass a space separated value in the query string like 
`api/countries/CA/places?params=St Catharines`
RIP will escape the white space in same way but this time my application will treat the `+` sign as white space as 
[W3.org](https://www.w3.org/Addressing/URL/uri-spec.html) under the title `Query String` suggest
> Within the query string, the plus sign is reserved as shorthand notation for a space. Therefore, real plus signs must be encoded.

So with RIP currently it is not possible to pass white space separated value as part of the URL path.
IMHO escape mechanism of RIP for `query string` is okay  but not for URL path.